### ISSUE TITLE
Squiz.NamingConventions.ValidVariableName.NotCamelCaps + property promotion = false positive

### ISSUE THREAD

**Issue by LastDragon-ru** [squizlabs/PHP_CodeSniffer#3564]
**Describe the bug**

Property promotions handling like variables, but they are class property

**Code sample**
```php
<?php declare(strict_types = 1);

namespace Tests;

class Test {
    public string $normal_property = 'abc';

    public function __construct(
        public string $promoted_property = 'abc',
    ) {
        // empty
    }
}

```

**Custom ruleset**
```xml
<?xml version="1.0"?>
<ruleset name="My Custom Standard">
  <rule ref="Squiz.NamingConventions.ValidVariableName.NotCamelCaps"/>
</ruleset>
```

**To reproduce**
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php ...`
3. See error message displayed
4. 
```
----------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------------
 9 | ERROR | Variable "promoted_property" is not in valid camel caps format
----------------------------------------------------------------------------
```

**Expected behavior**

No error.

**Versions (please complete the following information):**
 - OS: Windows 10
 - PHP: 8.0
 - PHPCS: 3.6.2
